source/server/server_factory_context_wrapper.h

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include "envoy/server/factory_context.h"
+
+namespace Envoy {
+namespace Server {
+
+// Wraps the global server context with a listener-local version, so that
+// downstream filters can access the downstream factory context.
+class ServerFactoryContextWrapper : public Configuration::ServerFactoryContext {
```

Comment:
If we are going to bikeshed maybe ListenerFactoryContextWrapper or something? I honestly find all of the different factory contexts very confusing at this point.